pkg/healthz/healthz_test.go

```diff
@@ -335,6 +355,14 @@ func (o *fakeResourceLock) Get(ctx context.Context) (*resourcelock.LeaderElectio
 		err := errors.NewNotFound(schema.ParseGroupResource("configmap"), "foo")
 		return nil, nil, err
 	}
+
+	// If simulating a remote-node leader, increment the LeaderTransitions field,
+	// simply to ensure a unique []byte representation each time.
+	// See the function documentation above for a fuller explanation.
+	if o.record.HolderIdentity == remoteIdentity {
+		o.record.LeaderTransitions++
+	}
```

Comment:
This is Tim's suggested change. 
And I added some locking too, in case multiple threads should happen to try to simultaneously update the state of the fake resource lock....although I don't think that is ever the case in the current tests.